common/namespace/handler.go

```diff
@@ -784,6 +797,48 @@ func (d *HandlerImpl) validateVisibilityArchivalURI(URIString string) error {
 	return archiver.ValidateURI(URI)
 }
 
+// maybeAppendToReplicationHistory adds an entry if the Namespace is becoming active in a new cluster.
+func (d *HandlerImpl) maybeAppendToReplicationHistory(
+	replicationHistory []*persistencespb.ReplicationStatus,
+	updateReplicationConfig *replicationpb.NamespaceReplicationConfig,
+	namespaceDetail *persistencespb.NamespaceDetail,
+) []*persistencespb.ReplicationStatus {
+	d.logger.Debug(
+		"maybeAppendToReplicationHistory",
+		tag.NewAnyTag("replicationHistory", replicationHistory),
+		tag.NewAnyTag("updateReplConfig", updateReplicationConfig),
+		tag.NewAnyTag("namespaceDetail", namespaceDetail),
+	)
+	desiredReplicationState := updateReplicationConfig.State
+	if desiredReplicationState == enumspb.REPLICATION_STATE_UNSPECIFIED {
+		desiredReplicationState = namespaceDetail.ReplicationConfig.State
+	}
+	// N.B., UNSPECIFIED is the same as NORMAL
+	if desiredReplicationState != enumspb.REPLICATION_STATE_NORMAL &&
+		desiredReplicationState != enumspb.REPLICATION_STATE_UNSPECIFIED {
+		d.logger.Debug("Replication state not NORMAL", tag.NewAnyTag("state", updateReplicationConfig.State))
+		return replicationHistory
+	}
+	newActiveCluster := updateReplicationConfig.ActiveClusterName
+	if newActiveCluster == "" {
+		newActiveCluster = namespaceDetail.ReplicationConfig.ActiveClusterName
+	}
+	lastActiveCluster := ""
+	if l := len(namespaceDetail.ReplicationHistory); l > 0 {
+		lastActiveCluster = namespaceDetail.ReplicationHistory[l-1].ActiveClusterName
+	}
+	if lastActiveCluster != newActiveCluster {
+		now := d.timeSource.Now()
+		replicationHistory = append(
```

Comment:
I think we should not allow this to grow unbounded. Probably keep small number of recent change records should be enough.

Reply:
Done